rest-api/tools/setup_database.sh

```diff
@@ -100,36 +91,50 @@ function finish {
 }
 trap finish EXIT
 
-echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
-     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
-     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
-     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
-     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
-     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
-     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
-     ' "db_user": "'$RDR_DB_USER'", '\
-     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
+function get_existing_db_config {
+  "Getting database info..."
+  tools/install_config.sh --key db_config --instance $INSTANCE \
+   --creds_file ${CREDS_FILE} --config_output "$TMP_DB_INFO_FILE"
+}
 
 for db_name in "rdr" "metrics"; do
   if [ "${UPDATE_PASSWORDS}" = "Y" ]
   then
-    cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
+    	echo "Updating database user passwords..."
+	randpw
+	ROOT_PASSWORD=$new_password
+	randpw
+	RDR_PASSWORD=$new_password
+	randpw
+	READONLY_PASSWORD=$new_password
+
+	echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
+	     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
+	     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
+	     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
+	     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
+	     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
+	     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
+	     ' "db_user": "'$RDR_DB_USER'", '\
+	     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
+
+	echo "Setting root password..."
+	#gcloud sql instances set-root-password $INSTANCE_NAME --password $ROOT_PASSWORD
+	gcloud sql users set-password root % --instance $INSTANCE_NAME --password $ROOT_PASSWORD
+
+        cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
   else
-    cat tools/create_db.sql | envsubst > $UPDATE_DB_FILE
+	get_existing_db_config
```

Comment:
If we aren't changing the existing config at all, then we don't need to run install_config.sh below, either.

Setting up the logic this way does give us the ability to re-run create_db.sql and have those changes applied to existing databases, which is nice. However, this logic won't give us the ability to update the existing db config without also changing the passwords. Ideally there would be a way of getting the existing db config and then creating a new db config based on the old password values, so you wouldn't have to change the passwords when adding new stuff to db_config. That said -- if you want to make it so that for now you must change passwords to update db_config, that's OK with me... just be aware we will have to do this in a maintenance windo win prod.

Reply:
If I understood you correctly, then this is done. Not sure what the use case is for updating config independently of creating/updating passwords.